docs/dev/libs/cep.md

```diff
@@ -246,63 +334,399 @@ pattern.where(event => ... /* some condition */).or(event => ... /* or condition
+                   
+              <p>E.g. a pattern like:</p>
+              {% highlight java %}
+              Pattern.<Event>begin("start").where(new SimpleCondition<Event>() {
+                @Override
+                public boolean filter(Event value) throws Exception {
+                  return value.getName().equals("c");
+                }
+              })
+              .followedBy("middle").where(new SimpleCondition<Event>() {
+                @Override
+                public boolean filter(Event value) throws Exception {
+                  return value.getName().equals("a");
+                }
+              }).oneOrMore().allowCombinations()
+              .followedBy("end1").where(new SimpleCondition<Event>() {
+                @Override
+                public boolean filter(Event value) throws Exception {
+                  return value.getName().equals("b");
+                }
+              });
+              {% endhighlight %}
+               <p>Will generate the following matches for an input sequence: C D A1 A2 A3 D A4 B</p>
+               
+               <p>with combinations enabled: {C A1 B}, {C A1 A2 B}, {C A1 A3 B}, {C A1 A4 B}, {C A1 A2 A3 B}, {C A1 A2 A4 B}, {C A1 A3 A4 B}, {C A1 A2 A3 A4 B}</p>
+               <p>without combinations enabled: {C A1 B}, {C A1 A2 B}, {C A1 A2 A3 B}, {C A1 A2 A3 A4 B}</p>
+       </td>
+       </tr>
+  </tbody>
+</table>
 </div>
 
 <div data-lang="scala" markdown="1">
+<table class="table table-bordered">
+    <thead>
+        <tr>
+            <th class="text-left" style="width: 25%">Pattern Operation</th>
+            <th class="text-center">Description</th>
+        </tr>
+    </thead>
+    <tbody>
+      
+        <tr>
+            <td><strong>where(condition)</strong></td>
+            <td>
+              <p>Defines a condition for the current state. Only if an event satisifes the condition, 
+              it can match the state. Multiple consecutive where() clauses lead to their condtions being 
+              ANDed:</p>
 {% highlight scala %}
-val strictNext: Pattern[Event, _] = start.next("middle")
+patternState.where(event => ... /* some condition */)
 {% endhighlight %}
+            </td>
+        </tr>
+        <tr>
+            <td><strong>or(condition)</strong></td>
+            <td>
+                <p>Adds a new condition which is ORed with an existing one. Only if an event passes one of the 
+                conditions, it can match the state:</p>
+{% highlight scala %}
+patternState.where(event => ... /* some condition */)
+    .or(event => ... /* alternative condition */)
+{% endhighlight %}
+                    </td>
+                </tr>
+       <tr>
+           <td><strong>subtype(subClass)</strong></td>
+           <td>
+               <p>Defines a subtype condition for the current pattern state. Only if an event is of this subtype, 
+               it can match the state:</p>
```

Comment:
You meant to say "Only if an event is of this subtype, can it match the state:", but I recommend

... An event can only match the state if it is of this subtype: